x-pack/plugins/uptime_monitoring/public/components/queries/filter_bar/filter_bar.tsx

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+// @ts-ignore No typings for EuiSearchBar
+import { EuiFlexGroup, EuiFlexItem, EuiIcon, EuiSearchBar, EuiToolTip } from '@elastic/eui';
+import { take } from 'lodash';
+import React from 'react';
+import { Query } from 'react-apollo';
+import { getFilterBarQuery } from './get_filter_bar';
+
+interface FilterBarProps {
+  dateRangeStart: number;
+  dateRangeEnd: number;
+  updateQuery: (query: object) => void;
+}
+
+const MAX_SELECTION_LENGTH = 20;
+const SEARCH_THRESHOLD = 2;
+
+export const FilterBar = ({ dateRangeEnd, dateRangeStart, updateQuery }: FilterBarProps) => (
+  <Query query={getFilterBarQuery} variables={{ dateRangeStart, dateRangeEnd }}>
+    {({ loading, error, data }) => {
+      if (loading) {
+        return 'Loading...';
+      }
+      if (error) {
+        return `Error ${error.message}`;
+      }
+      const {
+        filterBar: { port, id, scheme },
+      } = data;
+      const showFilterDisclaimer =
+        (id.length && id.length > MAX_SELECTION_LENGTH) ||
+        (port.length && port.length > MAX_SELECTION_LENGTH);
+      const filters = [
```

Comment:
Might be good to type these. Prevent regressions in the future

Reply:
I'd like to make a factory function too - I will add a TODO for that.